src/main/java/AddCommand.java

```diff
@@ -46,10 +48,15 @@ public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException
         default:
         }
 
+        if (result.equals("")) {
+            storage.writeToFile(tasks.saveList());
+            result = "Got it. I've added this task:\n"
+                    + t + "\n" + tasks.printSizeString();
+        }
         storage.writeToFile(tasks.saveList());
-        System.out.println("     Got it. I've added this task:");
-        System.out.println("       " + t);
-        tasks.printSize();
+        result = "Got it. I've added this task:\n"
+                + t + "\n" + tasks.printSizeString();
+        return result;
```

Comment:
Since you already have a UI class, maybe you can consider to put all related codes such as the code above included in UI class.